plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduModule.java

```diff
@@ -49,26 +59,37 @@ public KuduModule(TypeManager typeManager)
     }
 
     @Override
-    protected void configure()
+    protected void setup(Binder binder)
     {
-        bind(TypeManager.class).toInstance(typeManager);
+        binder.bind(TypeManager.class).toInstance(typeManager);
 
-        bind(KuduConnector.class).in(Scopes.SINGLETON);
-        bind(KuduMetadata.class).in(Scopes.SINGLETON);
-        bind(KuduTableProperties.class).in(Scopes.SINGLETON);
-        bind(ConnectorSplitManager.class).to(KuduSplitManager.class).in(Scopes.SINGLETON);
-        bind(ConnectorPageSourceProvider.class).to(KuduPageSourceProvider.class)
+        binder.bind(KuduConnector.class).in(Scopes.SINGLETON);
+        binder.bind(KuduMetadata.class).in(Scopes.SINGLETON);
+        binder.bind(KuduTableProperties.class).in(Scopes.SINGLETON);
+        binder.bind(ConnectorSplitManager.class).to(KuduSplitManager.class).in(Scopes.SINGLETON);
+        binder.bind(ConnectorPageSourceProvider.class).to(KuduPageSourceProvider.class)
                 .in(Scopes.SINGLETON);
-        bind(ConnectorPageSinkProvider.class).to(KuduPageSinkProvider.class).in(Scopes.SINGLETON);
-        bind(KuduHandleResolver.class).in(Scopes.SINGLETON);
-        bind(KuduSessionProperties.class).in(Scopes.SINGLETON);
-        bind(ConnectorNodePartitioningProvider.class).annotatedWith(ForClassLoaderSafe.class).to(KuduNodePartitioningProvider.class).in(Scopes.SINGLETON);
-        bind(ConnectorNodePartitioningProvider.class).to(ClassLoaderSafeNodePartitioningProvider.class).in(Scopes.SINGLETON);
-        bind(KuduRecordSetProvider.class).in(Scopes.SINGLETON);
-        configBinder(binder()).bindConfig(KuduClientConfig.class);
-
-        bind(RangePartitionProcedures.class).in(Scopes.SINGLETON);
-        Multibinder.newSetBinder(binder(), Procedure.class);
+        binder.bind(ConnectorPageSinkProvider.class).to(KuduPageSinkProvider.class).in(Scopes.SINGLETON);
+        binder.bind(KuduHandleResolver.class).in(Scopes.SINGLETON);
+        binder.bind(KuduSessionProperties.class).in(Scopes.SINGLETON);
+        binder.bind(ConnectorNodePartitioningProvider.class).annotatedWith(ForClassLoaderSafe.class).to(KuduNodePartitioningProvider.class).in(Scopes.SINGLETON);
+        binder.bind(ConnectorNodePartitioningProvider.class).to(ClassLoaderSafeNodePartitioningProvider.class).in(Scopes.SINGLETON);
+        binder.bind(KuduRecordSetProvider.class).in(Scopes.SINGLETON);
+        configBinder(binder).bindConfig(KuduClientConfig.class);
+        configBinder(binder).bindConfig(KuduAuthenticationConfig.class);
+
+        install(conditionalModule(
```

Comment:
Can we have this as a separate security module ?

Reply:
Split it out into a separate module, let me know if that is not what you meant.